main.tf

```diff
@@ -1445,6 +1445,42 @@ resource "aws_cloudwatch_event_target" "ec2_spot_instance_interruption" {
 
 
 
+resource "aws_cloudwatch_event_rule" "ecs_task_failed" {
+  depends_on = [aws_sns_topic_subscription.marbot]
+  count      = (var.ecs_task_failed && var.enabled) ? 1 : 0
+
+  name          = "marbot-basic-ecs-task-failed-${random_id.id8.hex}"
+  description   = "ECS Task failed. (created by marbot)"
+  tags          = var.tags
+  event_pattern = <<JSON
+{
+  "source": [ 
+    "aws.ecs"
+  ],
+  "detail-type": [
+    "ECS Task State Change"
+  ],
+  "detail": {
+    "lastStatus": ["STOPPED"],
+    "stopCode": [
+      "TaskFailedToStart",
+      "EssentialContainerExited"
```

Comment:
Isn't it possible that a container exits in a well-planned way?

Reply:
You are right.